### PR TITLE
Max in flight

### PR DESCRIPTION
**Pull request by evanfarrar**
__* A short explanation of the proposed change:__
** This exposes a new option for the deployment resource that 
__* An explanation of the use cases your change solves__
** Currently, rolling deploys deploy one instance at a time, but if an app has many instances this can take quite a while
* Links to any other associated PRs
** there is a CLI feature in progress, will link when it is a PR
* [x] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/CONTRIBUTING.md)

* [x] I have viewed, signed, and submitted the Contributor License Agreement

* [x] I have made this pull request to the `main` branch

* [x] I have run all the unit tests using `bundle exec rake`

* [ ] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/spec/README.md#cf-acceptance-tests-cats)

### CODE REVIEW

File: spec/unit/lib/cloud_controller/deployment_updater/updater_spec.rb [hunk 1]
```diff
@@ -218,6 +305,23 @@ module VCAP::CloudController
             subject.scale
           end.not_to(change(RouteMappingModel, :count))
         end
+
```
Comment (pivotalgeorge) [inline spec/unit/lib/cloud_controller/deployment_updater/updater_spec.rb:308]:
```suggestion

# when max_in_flight is greater than original_web_process_instance_count
```

File: spec/request/deployments_spec.rb
Comment (pivotalgeorge) [inline spec/request/deployments_spec.rb:1]: it seems we want to test a max_in_flight value other than `1` in this spec

**Review by pivotalgeorge — CHANGES_REQUESTED** [review]
deployment spec change

File: docs/v3/source/includes/resources/deployments/_header.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ Deployments are different than the traditional method of pushing app updates whi
 Deployment strategies supported:
 
 * [Rolling deployments](https://docs.cloudfoundry.org/devguide/deploy-apps/rolling-deploy.html) allows for
-applications to be deployed without incurring downtime by gradually rolling out instances.
+applications to be deployed without incurring downtime by gradually rolling out instances. Max-in-flight can be configured
```
Comment (pivotalgeorge) [inline docs/v3/source/includes/resources/deployments/_header.md:16]: sounde like we want to update the linked rolling-deploy doc with max-in-flight examples, but that's a separate story

File: spec/request/deployments_spec.rb
Comment (Samze) [reply spec/request/deployments_spec.rb:1]: Added

File: docs/v3/source/includes/resources/deployments/_header.md (hunk 2 above)
Comment (Samze) [reply docs/v3/source/includes/resources/deployments/_header.md:16]: Makes sense. Will have a follow up issue for this. Resolving for this PR.

**Review by pivotalgeorge — APPROVED** [review]
thanks for having a pair of pair of eyes on this with me @Samze !
Looks good to me with the deployment spec coverage

**Review by Samze — APPROVED** [review]

**Pull request merged**